src/framework/components/button/component.js

```diff
@@ -456,7 +456,7 @@ class ButtonComponent extends Component {
     }
 
     _applyTintImmediately(tintColor) {
-        if (!tintColor || !this._imageReference.hasComponent('element'))
+        if (!tintColor || !this._imageReference.hasComponent('element') || this._imageReference.entity.element.type != ELEMENTTYPE_IMAGE)
```

Comment:
Sorry for the delay in reviewing this, and thanks for this suggestion! There's a small problem with this approach which is that this would break the current behaviour of allowing you to link a `text` element to a button. I think you can change it to: ` || this._imageReference.entity.element.type === ELEMENTTYPE_GROUP` so that we keep that behaviour.

Reply:
I wasn't aware that was possible, have made this change.